doc/srv6/srv6_static_config_hld.md

```diff
@@ -0,0 +1,162 @@
+Specifically, there is no mechamism in SONiC allowing SDN controllers or users to directly add configuration for SRv6 without involving BGP.
+
+In this document, we first define a new **SRV6_MY_SID_TABLE** table in CONFIG_DB that serves as the configuration source of SRv6 in SONiC.
+Then, we design a new SRv6 Manager module in bgpcfgd to subscribe to the **SRV6_MY_SID_TABLE** table and compile changes in CONFIG_DB to changes in the configurations of FRR (Note: the new SRv6 Manager relies on the new configuration CLI brought in by [FRR PR#16894](https://github.com/FRRouting/frr/pull/16894)).
+To verify the correctness of the aforementioned flow, we also define the relevant YANG model specification.
+The workflow of the new mechanism is shown in the following diagram.
+
+![Static SRv6 Config flow](images/SRv6_bgpcfgd.png)
+
+The design details of each step is described in the following subsections.
+
+## 3.1 New Table in ConfigDB
+
+
+**SRV6_MY_SID_TABLE**
+
+Description: New table to hold local SID definition and SID to behavior mapping. (A simplified redefinition of SRV6_MY_SID_TABLE in [SRv6_HLD](./srv6_hld.md))
+
+Schema:
+
+```
+; New table
+; holds local SID to behavior mapping, allow 1:1 or n:1 mapping
+
+key = SRV6_MY_SID_TABLE|ipv6address
+; field = value
+block_len = blen             ; bit length of block portion in address, default 32
+node_len = nlen              ; bit length of node ID portion in address, default 16
+func_len = flen              ; bit length of function portion in address, default 16
+arg_len = alen               ; bit length of argument portion in address, default 0
+action = behavior            ; behaviors defined for the SID, default uN
+vrf = VRF_TABLE.key          ; Optional, VRF name for decapsulation actions, only applicable to "uDT6" and "uDT46" by now, default "default"
+
+For example:
+    "SRV6_MY_SID_TABLE" : {
+        "FCBB:BBBB:20::" : {
+           "action": "uN"
+        },
+        "FCBB:BBBB:20:F1::" : {
+           "action": "uDT46",
+        },
+        "FCBB:BBBB:20:F2::" : {
+           "action": "uDT46",
+           "vrf":  "VRF-1001"
+        },
+    }
+```
+
+
+## 3.2 Bgpcfgd changes
+
+To enable automatic programming SRv6 configurations from CONFIG_DB to FRR, we need to add a new module in bgpcfgd to watch changes in **SRV6_MY_SID_TABLE** and compile the corresponding changes in FRR's configurations.
+Following the naming convention of modules in bgpcfgd, we call this new module SRv6 Manager.
+The new SRv6 Manager are supposed to verify the validity of the configuration entries coming from the CONFIG_DB.
+If it gets an invalid configuration input, it should log the event in the syslog and not compile the configuration into FRR.
+
+## 3.3 YANG Model
+The simplified version of the YANG model is defined below.
+```
+module: sonic-srv6
+  +--rw sonic-srv6
+     +--rw SRV6_MY_SID
+     |  +--rw SRV6_MY_SID_LIST* [ip-address]
+     |     +--rw ip-address    inet:ipv6-address
```

Comment:
Can't we have same ipv6 address exist across VRFs? should use VRF as a key instead of field in the table?

Reply:
This table defines the MY_SID_LIST which should only have one instance in the current system. The VRF field is used to specify which VRF should be used for look up after the SRv6 functions decapsulates the packet. It is not used to specify which VRF the SID belongs to.